File: src/main/java/net/dontdrinkandroot/gitki/wicket/component/item/AdminDropdownItem.java
package net.dontdrinkandroot.gitki.wicket.component.item;

import net.dontdrinkandroot.gitki.model.Role;
import net.dontdrinkandroot.gitki.wicket.page.admin.ErrorTestPage;
import net.dontdrinkandroot.gitki.wicket.page.admin.LockListPage;
import net.dontdrinkandroot.gitki.wicket.security.Render;
import net.dontdrinkandroot.wicket.bootstrap.component.item.BookmarkablePageLinkItem;
import net.dontdrinkandroot.wicket.bootstrap.component.item.RepeatingDropdownItem;
import org.apache.wicket.markup.repeater.RepeatingView;
import org.apache.wicket.model.Model;
import org.apache.wicket.model.StringResourceModel;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@Render(Role.ADMIN)
public class AdminDropdownItem extends RepeatingDropdownItem<Void>
{
    public AdminDropdownItem(String id)
    {
        super(id, new StringResourceModel("gitki.administration"));
    }

    @Override
    protected void populateItems(RepeatingView itemView)
    {
        itemView.add(new UserListPageItem(itemView.newChildId()));
        itemView.add(new ConfigurationPageItem(itemView.newChildId()));
        itemView.add(new BookmarkablePageLinkItem(itemView.newChildId(), new StringResourceModel("gitki.locks"),
                LockListPage.class
        ));
        itemView.add(new BookmarkablePageLinkItem(itemView.newChildId(), Model.of("Errors"), ErrorTestPage.class));
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/component/item/AdminDropdownItem.kt
package net.dontdrinkandroot.gitki.wicket.component.item

import net.dontdrinkandroot.gitki.model.Role
import net.dontdrinkandroot.gitki.wicket.page.admin.ErrorTestPage
import net.dontdrinkandroot.gitki.wicket.page.admin.LockListPage
import net.dontdrinkandroot.gitki.wicket.security.Render
import net.dontdrinkandroot.wicket.bootstrap.component.item.ItemView
import net.dontdrinkandroot.wicket.bootstrap.component.item.RepeatingDropdownItem
import net.dontdrinkandroot.wicket.bootstrap.component.item.addPageLink
import net.dontdrinkandroot.wicket.model.localize
import org.apache.wicket.model.StringResourceModel

@Render(Role.ADMIN)
class AdminDropdownItem(id: String) :
    RepeatingDropdownItem<Void>(id, labelModel = StringResourceModel("gitki.administration")) {

    override fun populateItems(itemView: ItemView) {
        itemView.add(UserListPageItem(itemView.newChildId()))
        itemView.add(ConfigurationPageItem(itemView.newChildId()))
        itemView.addPageLink(localize("gitki.locks"), LockListPage::class.java)
        itemView.addPageLink("Errors", ErrorTestPage::class.java)
    }
}